crates/cli/src/utils/mod.rs

```diff
@@ -78,9 +78,36 @@ pub fn subscriber() {
         .init()
 }
 
-/// parse a hex str or decimal str as U256
-pub fn parse_u256(s: &str) -> Result<U256> {
-    Ok(if s.starts_with("0x") { U256::from_str(s)? } else { U256::from_dec_str(s)? })
+use thiserror::Error;
+
+#[derive(Debug, Error, PartialEq)]
+pub enum ParseU256Error {
+    #[error("Failed to parse as a decimal or hexadecimal value")]
+    ParseError,
+
+    #[error("Invalid format")]
+    InvalidFormat,
+}
+
+pub fn parse_u256(s: &str) -> Result<U256, ParseU256Error> {
+    // If decoding as a decimal fails, check if it has the "0x" prefix and try to decode as hex
+    if s.starts_with("0x") {
```

Comment:
I think we could do `strip_prefix("0x")` and check if it's `Some`, so we don't need to take the slice without the prefix manually